deps/rabbit/src/pg_local.erl

```diff
@@ -74,6 +76,12 @@ leave(Name, Pid) when is_pid(Pid) ->
     _ = ensure_started(),
     gen_server:cast(?MODULE, {leave, Name, Pid}).
 
+-spec count_unique_members(name()) -> non_neg_integer().
+
+count_unique_members(Name) ->
+    _ = ensure_started(),
+    ets:select_count(?TABLE, [{{{member, Name, '_'}, '_'}, [], [true]}]).
```

Comment:
I fear this operation might become too CPU costly with many connections.
I know this table is an `ordered_set` and the connections are located in O(log n). However, this operation still matches a large number of connections.

Specifically https://blog.rabbitmq.com/posts/2023/03/native-mqtt/ was able to connect 9 millions connections.
I would like to see how https://github.com/rabbitmq/mqtt-testing/tree/main/many-conn compares before and after this PR: That test case deploys 400 pods with 25k connections each (=10 million) where each pod connects every 80ms. So, there are
* 1000ms / 80ms = 12.5 connections per second per pod
* 400 pods * 12.5 connections per second = 5000 connections per second
* since this is a 3 node cluster: 5000 connections per second / 3 nodes = 1666 connections per second against each node
* during end of the test case there will be > 3,000,000 connections on each node, and therefore this operation will match 1666 * 3,000,000 = 5 billion records per second.

I've seen in some scaling tests that ETS / Mnesia operations become too CPU intense, and removed all such bottlenecks. Once such as example is https://github.com/rabbitmq/rabbitmq-server/commit/35afeffceb9044082bbb4f060922dc150d6d4250 . This PR should not introduce any new such bottlenecks. We do want to continue connecting millions of MQTT connections efficiently to RabbitMQ.

Alternatively, look into more efficient methods in determining the total number of connections per node.
For example https://github.com/rabbitmq/rabbitmq-server/blob/b4804a8015d1276f883365e335955d058d3f5c23/deps/rabbit/src/rabbit_vhost_limit.erl#L94 just reads a counter.

Another idea is simply `ets:info(tracked_connection, size).` which works for all native connections, that is AMQP 0.9.1, MQTT, Steams, and from 4.0 onwards AMQP 1.0 and STOMP.

There might be other more efficient ways to query the total number of connections per node.

Reply:
Hey, thanks for the review. I see what you mean with select vs simple size. 

First time I learn about rabbit_*tracking is from you today. Yet another registration facility it seems. 

The example of vhost_limit is a bit confusing at the moment since count_tracked_items_in does that rpc thingy I asked about on slack recently (what a coincidence!): https://github.com/rabbitmq/rabbitmq-server/blob/d258626148014f6463f48a5778c7ba00e3f74e30/deps/rabbit/src/rabbit_connection_tracking.erl#L161